### ISSUE TITLE
time data type validation

### ISSUE THREAD

**Issue by antoniosap** [balderdashy/waterline#835]
Hi all,

inserting a record in my model 'Crono':
...
  	timeStart: 	  { type: 'time' }, 
  	timeEnd: 	  { type: 'time' },  
...

....with js code ....
timeStart = new Date().getTime();
timeEnd = new Date().getTime();

give me validation error:

Invalid attributes sent to Crono:
 • timeStart
   • [object Object]
 • timeEnd
   • [object Object]

**Comment by gotmikhail** [comment]
https://github.com/balderdashy/waterline/issues/826

I reported this as a compatibility issue w/ `moment.js`. Looks like it might actually be a compatibility issue w/ the js `Date` object.

**Comment by gsdatta** [comment]
Getting the same issue, with postgres adapter if that helps.

**Comment by antoniosap** [comment]
also redis adapter

**Comment by gotmikhail** [comment]
Mine was w/ the mysql adapter

**Comment by gsdatta** [comment]
Looks like it's not even an issue with the Date object. I tried using moment to parse the input value (rather than Date.parse()), and returned the Date representation of the object if the moment object was valid. Even with what I'm sure is a valid object, I'm receiving the same validation error. 

```
Error (E_VALIDATION) :: 2 attributes are invalid
    at WLValidationError.WLError (/Volumes/Storage/Development/myproject/node_modules/waterline/lib/waterline/error/WLError.js:33:18)
    at new WLValidationError (/Volumes/Storage/Development/myproject/node_modules/waterline/lib/waterline/error/WLValidationError.js:20:28)
    at /Volumes/Storage/Development/myproject/node_modules/waterline/lib/waterline/query/validate.js:45:43
    at allValidationsChecked (/Volumes/Storage/Development/myproject/node_modules/waterline/lib/waterline/core/validations.js:203:5)
    at done (/Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:135:19)
    at /Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:32:16
    at validate (/Volumes/Storage/Development/myproject/node_modules/waterline/lib/waterline/core/validations.js:135:49)
    at /Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:125:13
    at Array.forEach (native)
    at _each (/Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:46:24)
    at Object.async.each (/Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:124:9)
    at Validator.validate (/Volumes/Storage/Development/myproject/node_modules/waterline/lib/waterline/core/validations.js:200:9)
    at async.series.runner (/Volumes/Storage/Development/myproject/node_modules/waterline/lib/waterline/query/validate.js:41:25)
    at /Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:610:21
    at /Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:249:17
    at iterate (/Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:149:13)
    at /Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:160:25
    at /Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:251:21
    at /Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:615:34
    at /Volumes/Storage/Development/myproject/node_modules/waterline/lib/waterline/query/validate.js:35:11
    at /Volumes/Storage/Development/myproject/node_modules/waterline/node_modules/async/lib/async.js:157:25
    at /Volumes/Storage/Development/myproject/node_modules/waterline/lib/waterline/query/validate.js:29:13
```

**Comment by gsdatta** [comment]
Found the issue! In the validations, there was no validation defined for time, so it would fail no matter what the input was. I've created a fix which includes a typecast for time specifically, though it may need to tweaked a bit. I also added a time validator to the validation list.

I'm using moment.js for the fact that it supports much better validity checking and better parsing features. I'm submitting a PR to this repo with the additional time typecast, and submitted a PR to anchor for the additional validation. 

For those of you who want the fix, the change I made to anchor was the following in the file: anchor/lib/match/rules.js. This can be found under waterlines node_modules folder. The change was:

```javascript
var moment = require('moment');
...
'time': function(x) {return moment(x, "hh:mm:ssa").isValid();},
```

**Comment by dmarcelino** [comment]
For reference, the proposed fix is PR #863.

**Comment by dmarcelino** [comment]
For reference, there is also a big discussion about date/time formates and handling in balderdashy/sails-postgresql#153.

**Comment by sailsbot** [comment]
Thanks for posting, @antoniosap.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
